configs/tc.prod.yml

```diff
@@ -1,9 +1,154 @@
+  best-model: chrf
+
+  # Use the Opus Cleaner tool on the data cleaning step.
+  # https://github.com/hplt-project/OpusCleaner
+  use-opuscleaner: "true"
+
+  # Bicleaner is a tool that aims at detecting noisy sentence pairs in a parallel corpus.
+  # It indicates the likelihood of a pair of sentences being mutual translations (with a
+  # value near to 1) or not (with a value near to 0). Sentence pairs considered very
+  # noisy are scored with 0. Bicleaner AI will be used first if the language is
+  # available, otherwise it will fallback to the original non-AI Bicleaner.
+  #
+  # See:
+  #   https://github.com/bitextor/bicleaner-ai
+  #   https://github.com/bitextor/bicleaner
+  #
+  # For supported languages see:
+  #   https://github.com/bitextor/bicleaner-ai-data/releases
+  #   https://github.com/bitextor/bicleaner-data/releases
+  #
+  # New language releases are added to: taskcluster/ci/fetch/bicleaner.yml
+  bicleaner:
+    default-threshold: 0.5
+    dataset-thresholds:
+      opus_CCAligned/v1: 0.7
+      opus_LinguaTools-WikiTitles/v2014: 0.7
+      opus_OpenSubtitles/v2018: 0.8             # Example of a higher filtering level.
+      opus_ParaCrawl/v9: 0.0                    # Example of cleaning being disabled.
+      opus_WikiMatrix/v1: 0.7
+      opus_bible-uedin/v1: 0.7
+
+  # Limits the maximum sentences use in the monolingual data for both the source and target
+  # languages.
+  mono-max-sentences-src: 100_000_000
+  mono-max-sentences-trg: 20_000_000
+
+  # The number of sentences split out into separate files for the monolingual data.
+  # Also see taskcluster.split-chunks below.
+  split-length: 2_000_000
+
+  # How many bytes of the corpus are sampled to be used in SentencePiece
+  # tokenization training.
+  spm-sample-size: 10_000_000
+
+  # How big the vocab will be. The vocab is shared between the source and the
+  # target languages.
+  #
+  # In "Finding the Optimal Vocabulary Size for Neural Machine Translation", the
+  # researchers tests vocab sizes between 500 and 64,000. A general value table
+  # depends on the vocabulary size.
+  #
+  # | Corpus Size | Optimal vocab size |
+  # | ----------- | ------------------ |
+  # |   4,500,000 |    32,000 - 64,000 |
+  # |   1,000,000 |              8,000 |
+  # |     500,000 |     4,000 - 16,000 |
+  # |      30,000 |              1,000 |
+  #
+  # See section 4 for their conclusions.
+  #   https://aclanthology.org/2020.findings-emnlp.352.pdf
+  #
+  # The larger the vocab, the slower the training and the inference will be as each
+  # token in the vocab contributes to the amount of probabilities that will need
+  # to be generated when making a token prediction. This is a trade off on the quality
+  # of the translation, and the performance.
+  spm-vocab-size: 32_000
+
+  # Determine how many teachers to train. The teachers will be identical except they
```

Comment:
I think the better place for the extended docs would be in the training guide in the /docs. I would recommend limiting comments here to one liners with a simple description of the parameter, because we often use this config as a template for the new runs and it will be harder to navigate with all the embedded docs.

Reply:
Sounds good. I think a paragraph is still fine, but something this big can live in the docs folder.